From d2c76214e625f41bb8e32d7808f8223c927a2c3f Mon Sep 17 00:00:00 2001
From: Ran Wang <ran.wang_1@nxp.com>
Date: Wed, 16 Jan 2019 15:40:33 +0800
Subject: [PATCH 425/741] arm64: dts: Fix DWC3 IP VBUS glitch issue on
 Layerscape platforms

commit 70f130039a7fcca5b2315157ecbe9f984a1549f8 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Cover LS1012A, LS1043A, LS1046A, LS1088A, LS208xA, LX2160A

Signed-off-by: Ran Wang <ran.wang_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi | 1 +
 arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi | 3 +++
 arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi | 3 +++
 arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi | 2 ++
 arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi | 2 ++
 arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi | 2 ++
 6 files changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
index 75f364acb40c..a7d8f90767d1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
@@ -442,6 +442,7 @@
 			snps,quirk-frame-length-adjustment = <0x20>;
 			snps,dis_rxdet_inp3_quirk;
 			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+			snps,host-vbus-glitches;
 		};
 
 		sata: sata@3200000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index 6fc1cffcfcb9..94bc58a4a30d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -711,6 +711,7 @@
 				usb3-lpm-capable;
 				snps,dis-u1u2-when-u3-quirk;
 				snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+				snps,host-vbus-glitches;
 				configure-gfladj;
 			};
 
@@ -724,6 +725,7 @@
 				usb3-lpm-capable;
 				snps,dis-u1u2-when-u3-quirk;
 				snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+				snps,host-vbus-glitches;
 				configure-gfladj;
 			};
 
@@ -737,6 +739,7 @@
 				usb3-lpm-capable;
 				snps,dis-u1u2-when-u3-quirk;
 				snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+				snps,host-vbus-glitches;
 				configure-gfladj;
 			};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
index 64796cb1905e..c21586ad852b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
@@ -608,6 +608,7 @@
                                snps,dis_rxdet_inp3_quirk;
                                usb3-lpm-capable;
                                snps,dis-u1u2-when-u3-quirk;
+			       snps,host-vbus-glitches;
                        };
  
                        usb1: usb@3000000 {
@@ -619,6 +620,7 @@
                                snps,dis_rxdet_inp3_quirk;
                                usb3-lpm-capable;
                                snps,dis-u1u2-when-u3-quirk;
+                               snps,host-vbus-glitches;
                        };
 
                        usb2: usb@3100000 {
@@ -630,6 +632,7 @@
                                snps,dis_rxdet_inp3_quirk;
                                usb3-lpm-capable;
                                snps,dis-u1u2-when-u3-quirk;
+                               snps,host-vbus-glitches;
                        };
 
                        sata: sata@3200000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
index 2fab967454ad..70c94257c7ba 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
@@ -429,6 +429,7 @@
 			snps,dis_rxdet_inp3_quirk;
 			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
 			status = "disabled";
+                        snps,host-vbus-glitches;
 		};
 
 		usb1: usb3@3110000 {
@@ -440,6 +441,7 @@
 			snps,dis_rxdet_inp3_quirk;
 			status = "disabled";
 			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+			snps,host-vbus-glitches;
 		};
 
 		sata: sata@3200000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
index 6dc8c8a51d4b..4f906d3b8642 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
@@ -811,6 +811,7 @@
 			snps,quirk-frame-length-adjustment = <0x20>;
 			snps,dis_rxdet_inp3_quirk;
 			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+			snps,host-vbus-glitches;
 		};
 
 		usb1: usb3@3110000 {
@@ -822,6 +823,7 @@
 			snps,quirk-frame-length-adjustment = <0x20>;
 			snps,dis_rxdet_inp3_quirk;
 			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+			snps,host-vbus-glitches;
 		};
 
 		serdes1: serdes@1ea0000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi b/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi
index fba6ff3e38de..33ee6f822567 100644
--- a/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi
@@ -845,6 +845,7 @@
 			snps,quirk-frame-length-adjustment = <0x20>;
 			snps,dis_rxdet_inp3_quirk;
 			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+			snps,host-vbus-glitches;
 			status = "disabled";
 		};
 
@@ -856,6 +857,7 @@
 			snps,quirk-frame-length-adjustment = <0x20>;
 			snps,dis_rxdet_inp3_quirk;
 			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+			snps,host-vbus-glitches;
 			status = "disabled";
 		};
 
-- 
2.17.1

